Server/mods/deathmatch/logic/luadefs/CLuaWorldDefs.cpp

```diff
@@ -1433,3 +1436,45 @@ int CLuaWorldDefs::getOcclusionsEnabled(lua_State* luaVM)
     lua_pushboolean(luaVM, false);
     return 1;
 }
+
+int CLuaWorldDefs::RequestModel(lua_State* luaVM)
+{
+    ushort usModelID = 0;
+    eModelType modelType;
+
+    CScriptArgReader argStream(luaVM);
+    argStream.ReadNumber(usModelID);
+    argStream.ReadEnumString(modelType);
+
+    if (!argStream.HasErrors())
+    {
+        ushort usParentID = -1;
+        if (argStream.NextIsNumber())
+            argStream.ReadNumber(usParentID);
+        else
+        {
+            switch (modelType)
+            {
+            case eModelType::PED:
+                usParentID = 7; // male01
+                break;
+            case eModelType::OBJECT:
+                usParentID = 1337; // BinNt07_LA (trash can)
+                break;
+            case eModelType::VEHICLE:
+                usParentID = VT_LANDSTAL;
+                break;
+            default:
+                break;
+            }
+        }
+
+        lua_pushboolean(luaVM, g_pGame->GetModelManager()->RequestModel(usModelID, usParentID, modelType));
+        return 1;
+    }
+    else
+        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
+
+    lua_pushboolean(luaVM, false);
+    return 1;
+}
```

Comment:
Use the new parser please.

Reply:
I think new parser don't support potential optional args for model settings per model type

Reply:
`std::optional`?

Reply:
`std::optional` and `std::variant` works bad with numeric types (`std::variant<float,int>`) and old parser will be readable here i think

Reply:
What do you mean by "works bad"?

Reply:
`std::variant<float,int>` will accept any number value as float and `std::variant<int, float>` as int (Lua 1.3333 -> C++ 1)

Reply:
Well, yes, of course it won't chose the more *fitting* type, but why would you need such a feature anyways? You either accept an int or a float, not both. IMHO.